com.unity.render-pipelines.core/Runtime/Inputs/InputRegistering.cs

```diff
@@ -76,18 +149,39 @@ public static void RegisterInputs(List<InputManagerEntry> entries)
 
             var inputManager = assets[0];
 
-            // Wrap in serialized object
+            // Wrap in serialized object to access c++ fields
             var soInputManager = new SerializedObject(inputManager);
             var spAxes = soInputManager.FindProperty("m_Axes");
 
-            foreach (InputManagerEntry entry in entries)
-            {
-                WriteEntry(spAxes, entry);
-            }
+            // At this point, we assume that entries in spAxes are already unique.
+            
+            // Ensure no double entry are tried to be registered (trim early)
+            MakeUniquePendingInputsToRegister();
+
+            // Cache already existing entries to minimaly use serialization
+            var cachedEntries = GetCachedInputs(spAxes);
+            // And trim pending entries regarding already cached ones.
+            RemovePendingInputsToAddThatAreAlreadyRegistered(cachedEntries, s_PendingInputsToRegister);
+
+            // Add now unique entries
+            AddEntriesWithoutCheck(spAxes, s_PendingInputsToRegister);
 
             // Commit
             soInputManager.ApplyModifiedProperties();
         }
+
+        public static void RegisterInputs(List<InputManagerEntry> entries)
+        {
+#if ENABLE_INPUT_SYSTEM && ENABLE_INPUT_SYSTEM_PACKAGE
+            Debug.LogWarning("Trying to add entry in the legacy InputManager but using InputSystem package. Skiping.");
+            return;
+#else
+            s_PendingInputsToRegister.AddRange(entries);
+
+            //delay the call in order to do only one pass event if several different class register inputs
+            EditorApplication.delayCall += DelayedRegisterInput;
```

Comment:
This line is replaced by `DelayedRegisterInput();` in the test to compare algorithm in same conditions.